frontend/packages/console-shared/src/hooks/useServicesWatcher.ts

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getServicesForResource } from '../utils';
+
+export const useServicesWatcher = (
+  resource: K8sResourceKind,
+): { loaded: boolean; loadError: string; services: K8sResourceKind[] } => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [services, setServices] = React.useState<K8sResourceKind[]>();
+  const { namespace } = resource.metadata;
+  const watchedResources = React.useMemo(
+    () => ({
+      services: {
+        isList: true,
+        kind: 'Service',
+        namespace,
+      },
+    }),
+    [namespace],
+  );
+
+  const resources = useK8sWatchResources(watchedResources);
```

Comment:
Can we simplify the error code below when using `useK8sWatchResource` instead of `useK8sWatchResources` here?

```suggestion
  const result = useK8sWatchResource(watchedResources);
  return {
    services: result[0],
    loaded: result[1],
    loadError, result[2],
  }
```